servers/src/grin/i2p.rs

```diff
@@ -53,7 +56,12 @@ pub fn init(config: &mut ServerConfig) -> Option<Arc<Session>> {
 }
 
 fn start_i2pd() {
-	unimplemented!()
+	let i2pd_path = which::which("i2pd").unwrap();
+	Command::new(i2pd_path.to_str().unwrap())
+		.spawn()
+		.expect("i2pd failed to start. Ensure i2pd is on your system path.");
+	// allow i2pd a moment to startup before attempting SAM connection
+	std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
Can't you retry the SAM connection for some time instead of relying on the sleep?